python/lsst/dbb/buffmngrs/endpoint/ingester/ingester.py

```diff
@@ -226,17 +214,17 @@ def _fetch(self):
         #         (SELECT *
         #         FROM <event table> AS e
         #         WHERE e.files_id = f.id);
+        query = self.session.query(self.File.id).\
+            filter(~exists().where(self.Event.files_id == self.File.id))
         try:
-            query = self.session.query(self.File.id).\
-                filter(~exists().where(self.Event.files_id == self.File.id))
-        except Exception as ex:
-            logger.error(f"failed to check for new files: {ex}")
-        else:
             for (id_,) in query:
                 rec = self.Event(status=Status.UNTRIED.value,
                                  start_time=datetime.datetime.now(),
                                  files_id=id_)
                 self.session.add(rec)
+        except Exception as ex:
+            logger.error(f"failed to check for new files: {ex}")
```

Comment:
With the scope of this new try clause, can the exception also be due to problem with the add? (i.e., "check" sounds like the query instead of inserting into table).  (And generic Exception instead of SQLAlchemy ...)

Reply:
Honestly, I'm not 100% sure, but to my current understanding of how lazy evaluation works in SQLAlchemy, I'd say: "Probably no." ;)

The official documentation of [add()](https://docs.sqlalchemy.org/en/13/orm/session_api.html#sqlalchemy.orm.session.Session.add) doesn't seem to indicate if it raises any exception either.